game/scripts/vscripts/items/bottle.lua

```diff
@@ -0,0 +1,51 @@
+LinkLuaModifier( "modifier_bottle_regeneration", "items/bottle.lua", LUA_MODIFIER_MOTION_NONE )
+
+--------------------------------------------------------------------------------
+
+item_bottle = class({})
+
+function item_bottle:OnSpellStart()
+	local restore_time = self:GetSpecialValueFor( "restore_time" )
+
+	self:GetCaster():EmitSoundParams( "Bottle.Drink", 0, 0.5, 0 )
+
+	self:GetCaster():AddNewModifier( self:GetCaster(), self, "modifier_bottle_regeneration", { duration = restore_time } )
+
+	self:SetCurrentCharges( self:GetCurrentCharges() - 1 )
+end
+
+--------------------------------------------------------------------------------
+
+modifier_bottle_regeneration = class({})
+
+function modifier_bottle_regeneration:OnCreated()
+	self.health_restore = self:GetAbility():GetSpecialValueFor( "health_restore" )
+	self.mana_restore = self:GetAbility():GetSpecialValueFor( "mana_restore" )
+	self.restore_time = self:GetAbility():GetSpecialValueFor( "restore_time" )
+end
+
+function modifier_bottle_regeneration:DeclareFunctions()
+	local funcs = {
+		MODIFIER_PROPERTY_HEALTH_REGEN_CONSTANT,
+		MODIFIER_PROPERTY_MANA_REGEN_CONSTANT,
```

Comment:
Needs to listen on taking damage so that you can cancel bottle regen when you take hero damage. Bottles aren't intended to be good against heroes mid-fight.